ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java

```diff
@@ -501,25 +480,46 @@ public CompletableFuture<DeleteResult> delete(String accountName, String contain
   }
 
   @Override
-  public CompletableFuture<List<StaleNamedBlob>> pullStaleBlobs() {
+  public CompletableFuture<StaleBlobsWithLatestBlobName> pullStaleBlobs(Container container, String blobName) {
     TransactionStateTracker transactionStateTracker =
         new GetTransactionStateTracker(remoteDatacenters, localDatacenter);
     return executeGenericTransactionAsync(true, (connection) -> {
       long startTime = this.time.milliseconds();
-      List<StaleNamedBlob> staleNamedBlobResults = runPullStaleBlobs(connection);
+      List<StaleNamedBlob> staleNamedBlobResults;
+      StaleBlobsWithLatestBlobName staleBlobsWithLatestBlobName = null;
+      List<StaleNamedBlob> potentialStaleNamedBlobResults = getAllBlobsForContainer(connection, container, blobName);
+
+      int resultSize = potentialStaleNamedBlobResults.size();
+      Container.ContainerStatus status = container.getStatus();
+
+      if (resultSize == config.queryStaleDataMaxResults) {
+        if (status == Container.ContainerStatus.ACTIVE) {
+          staleBlobsWithLatestBlobName = getStaleBlobsForActiveContainer(potentialStaleNamedBlobResults, config.staleDataRetentionDays);
+          return staleBlobsWithLatestBlobName;
+        } else if (status == Container.ContainerStatus.INACTIVE) {
+          staleNamedBlobResults = potentialStaleNamedBlobResults;
+          return new StaleBlobsWithLatestBlobName(staleNamedBlobResults, "\0");
```

Comment:
it should be the last blob name so next time we can continue from the last blob name.